### PR TITLE
New badges

### PR DESCRIPTION
**Pull request by jrlambs**
- sort channels by protocol, operation, and name
- split operation badge from protocol badge
- add icon to operation badge

### CODE REVIEW

File: src/app/channels/channels.component.css [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+.publish-badge {
+    background-color: #9BD279;
+}
+
+.protocol-badge {
+    background-color: #347AEB;
+    color: #fff
+}
+
+.fa {
```
Comment (timonback) [inline src/app/channels/channels.component.css]: Such a generic class will apply to font-awesome icons on the page.
I suggest making it a bit more selective

```suggestion
.badge .fa {
```

File: src/app/channels/channels.component.ts [hunk 2]
```diff
@@ -20,8 +20,30 @@ export class ChannelsComponent implements OnInit {
   ngOnInit(): void {
     this.nameSubscription = this.asyncApiService.getCurrentAsyncApiName().subscribe(name => {
       this.docName = name;
-      this.asyncApiService.getAsyncApis().subscribe(asyncapi => this.channels = asyncapi.get(name).channels);
+      this.asyncApiService.getAsyncApis().subscribe(asyncapi => {
+        this.channels = this.sortChannels(asyncapi.get(name).channels);
+        console.log(this.channels);
```
Comment (timonback) [inline src/app/channels/channels.component.ts]:
```suggestion
```

Probably not needed anymore :)

File: src/app/channels/channels.component.ts [hunk 3]
```diff
@@ -20,8 +20,30 @@ export class ChannelsComponent implements OnInit {
...
+          return a?.operation?.operation?.localeCompare(b?.operation?.operation);
+        }
+      } else {
+        return a?.operation?.protocol?.localeCompare(b?.operation?.protocol);
+      }
+    });
+  }
+
+
+  private sortByProperty(toSort: Array<any>, property: string) {
```
Comment (timonback) [inline src/app/channels/channels.component.ts]: Is this used anywhere?

File: src/app/shared/asyncapi.service.ts [hunk 4]
```diff
@@ -123,6 +123,8 @@ export class AsyncApiService {
     private mapOperation(operationName: string, message: Message, bindings?: any): Operation {
         return {
             type: this.getProtocol(bindings) + operationName,
```
Comment (timonback) [inline src/app/shared/asyncapi.service.ts:125]: As the `type` got split, is it possible to remove the field?

**Review by timonback — APPROVED** [review]
LGTM. I added some minor style items

**Comment by timonback** [comment]
Closes https://github.com/springwolf/springwolf-ui/pull/20

File: src/app/channels/channels.component.ts (hunk 3 above)
Comment (jrlambs) [reply src/app/channels/channels.component.ts]: nope, gone now.

File: src/app/shared/asyncapi.service.ts (hunk 4 above)
Comment (jrlambs) [reply src/app/shared/asyncapi.service.ts:125]: yep, gone now.

**Comment by stavshamir** [comment]
Looks great, thanks

**Pull request merged**